packages/php-db-import-export/tests/functional/Bigquery/StageImportTest.php

```diff
@@ -380,4 +380,176 @@ public function testLoadFromTableWithNullValueToRequiredColumn(): void
+    `first_name` STRING(100),
+    `last_name` STRING(100)
+);',
+                BigqueryQuote::quoteSingleIdentifier(self::TEST_DATABASE),
+                BigqueryQuote::quoteSingleIdentifier(self::TABLE_GENERIC),
+            ),
+        );
+        $this->bqClient->runQuery($query);
+
+        $query = $this->bqClient->query(
+            sprintf(
+                'CREATE TABLE %s.%s (
+            `id` INTEGER,
+    `first_name` STRING(3),
+    `last_name` STRING(3)
+);',
+                BigqueryQuote::quoteSingleIdentifier(self::TEST_DATABASE),
+                BigqueryQuote::quoteSingleIdentifier(self::TABLE_GENERIC.'_DEST'),
+            ),
+        );
+        $this->bqClient->runQuery($query);
+        $this->bqClient->runQuery($this->bqClient->query(
+            sprintf(
+                'INSERT INTO %s.%s(`id`,`first_name`,`last_name`) VALUES (1,\'test\',\'test\')',
+                BigqueryQuote::quoteSingleIdentifier(self::TEST_DATABASE),
+                BigqueryQuote::quoteSingleIdentifier(self::TABLE_GENERIC),
+            ),
+        ));
+
+        $importer = new ToStageImporter($this->bqClient);
+        $ref = new BigqueryTableReflection(
+            $this->bqClient,
+            self::TEST_DATABASE,
+            self::TABLE_GENERIC.'_DEST',
+        );
+
+        $this->expectException(BigqueryException::class);
+        $this->expectExceptionMessage('Field first_name: STRING(3) has maximum length 3 but got a value with length 4');
+        $importer->importToStagingTable(
+            new Table(
+                self::TEST_DATABASE,
+                self::TABLE_GENERIC,
+                [],
+                [],
+            ),
+            $ref->getTableDefinition(),
+            new BigqueryImportOptions(
+                usingTypes: BigqueryImportOptions::USING_TYPES_USER,
+            ),
+        );
+    }
+
+    public function testTableToTableImportFromTableLengthFitInLowerLength(): void
+    {
+        $query = $this->bqClient->query(
+            sprintf(
+                'CREATE TABLE %s.%s (
+            `id` INTEGER,
+    `first_name` STRING(100),
+    `last_name` STRING(100)
+);',
+                BigqueryQuote::quoteSingleIdentifier(self::TEST_DATABASE),
+                BigqueryQuote::quoteSingleIdentifier(self::TABLE_GENERIC),
+            ),
+        );
+        $this->bqClient->runQuery($query);
+
+        $query = $this->bqClient->query(
+            sprintf(
+                'CREATE TABLE %s.%s (
+            `id` INTEGER,
+    `first_name` STRING(5),
+    `last_name` STRING(5)
+);',
+                BigqueryQuote::quoteSingleIdentifier(self::TEST_DATABASE),
+                BigqueryQuote::quoteSingleIdentifier(self::TABLE_GENERIC.'_DEST'),
+            ),
+        );
+        $this->bqClient->runQuery($query);
+        $this->bqClient->runQuery($this->bqClient->query(
+            sprintf(
+                'INSERT INTO %s.%s(`id`,`first_name`,`last_name`) VALUES (1,\'test\',\'test\')',
```

Comment:
Když sa data vejdů tak to BQ klidně naimportuje takže bych to nechal tak ať si stím poradí a neomezoval to.